jdaviz/core/template_mixin.py

```diff
@@ -1677,6 +1679,82 @@ def _check_dataset_spectral_subset_valid(self, event={}, return_ranges=False):
+        that correspond to a non-finite uncertainty at that index.
+
+        In model fitting, the presence of finite, fittable data with corresponding
+        non-finite uncertainties can cause issues. Finite data values will be
+        filtered out in this case which may be undesirable. This traitlet when
+        True triggers a warning in the model fitting plugin (in specviz only,
+        currently) if there are any finite values with non-finite uncertainties.
+
+        Note that if a the uncertainty array is FULLY non-finite and the data is
+        FULLY finite, uncertainties will be set to None (in the Specviz parser),
+        so this traitlet will be False in that case (and therefore no warning
+        message displayed in the plugin).
+    """
+
+    non_finite_uncertainty_mismatch = Bool(False).tag(sync=True)
+
+    # every time a data/subset selection is changed, check the data selection and
+    # its uncertainties to see if there are any finite data elements with
+    # uncertainties. Warn in plugin if this occurs.
+
+    # note! the 'n/a' traitlet doesn't exist. we are watching it here because
+    # DatasetSpectralSubsetValidMixin is also watching the exact set
+    # of real traitlets, which creates a bug where only the last inhereted method
+    # will run, so we need to spoof it like this.
+    @observe("n/a", "dataset_selected", "spectral_subset_selected")
+    def _check_non_finite_uncertainty_mismatch(self, event={}):
+
+        if not hasattr(self, 'dataset') or self.dataset_selected == '':
+            # during initial init, this can trigger before the component is initialized
+            return
+
+        if self.spectral_subset_selected == "Entire Spectrum":
+            if not hasattr(self, '_get_1d_spectrum'):
+                # only model_fitting has _get_1d_spectrum(), but this method is here
+                # instead of there  because it may eventually be used by other plugins.
+                # if that happens, move _get_1d_spectrum() somewhere more general
+                m = "_get_1d_spectrum() must be available in plugin to use"
+                " NonFiniteUncertaintyMismatchMixin"
+                raise NotImplementedError(m)
+            spec = self._get_1d_spectrum()  # this is a method in model fitting
+            flux = spec.flux
+            uncert = spec.uncertainty
+
+            if uncert is None:
+                self.non_finite_uncertainty_mismatch = False
+                return self.non_finite_uncertainty_mismatch
+        else:
+            # get selected subset
+            spec = self._apply_subset_masks(self._get_1d_spectrum(), self.spectral_subset)
+            flux = spec.flux[~spec.mask]
+            uncert = spec.uncertainty
+            if uncert is None:
+                self.non_finite_uncertainty_mismatch = False
+                return self.non_finite_uncertainty_mismatch
+            uncert = spec.uncertainty[~spec.mask]
```

Comment:
We need the hasattr check regardless of the spectral subset, and moving around the order should also allow for avoiding code duplication (this isn't tested and indentation might not quite be right).

```suggestion
        if not hasattr(self, '_get_1d_spectrum'):
            # only model_fitting has _get_1d_spectrum(), but this method is here
            # instead of there  because it may eventually be used by other plugins.
            # if that happens, move _get_1d_spectrum() somewhere more general
            m = "_get_1d_spectrum() must be available in plugin to use"
            " NonFiniteUncertaintyMismatchMixin"
            raise NotImplementedError(m)

        spec = self._get_1d_spectrum()  # this is a method in model fitting
        if spec.uncertaity is None:
            self.non_finite_uncertainty_mismatch = False
            return self.non_finite_uncertainty_mismatch

        if self.spectral_subset_selected != "Entire Spectrum":
            # get selected subset
            spec = self._apply_subset_masks(spec, self.spectral_subset)
            flux = spec.flux[~spec.mask]
            uncert = spec.uncertainty[~spec.mask]
```

Reply:
i think this works, the check for None works before we determine if a subset was selected because the only way a subset's uncertainties can be None is if the whole spectrum's uncertainty is None. I can also remove the check for none before i apply the subset mask because the same logic applies there.